engine/src/main/java/org/camunda/bpm/engine/impl/cfg/ProcessEngineConfigurationImpl.java

```diff
@@ -1681,11 +1688,33 @@ protected void initDataSource() {
       }
     }
 
+    checkTransactionIsolationLevel();
+
     if (databaseType == null) {
       initDatabaseType();
     }
   }
 
+  protected void checkTransactionIsolationLevel() {
+    Integer currentIsolationLevel = getCurrentTransactionIsolationLevel();
+    String isolationLevelName = ISOLATION_LEVEL_CONSTANT_NAMES.get(currentIsolationLevel);
+    if (currentIsolationLevel != Connection.TRANSACTION_READ_COMMITTED) {
+      if (skipIsolationLevelCheck) {
+        LOG.logSkippedIsolationLevelCheck(isolationLevelName);
+      } else {
+        throw LOG.invalidTransactionIsolationLevel(isolationLevelName);
+      }
+    }
+  }
+
+  protected Integer getCurrentTransactionIsolationLevel() {
+    try (Connection connection = dataSource.getConnection()) {
+      return connection.getTransactionIsolation();
+    } catch (SQLException e) {
+      throw LOG.databaseConnectionAccessException(e);
```

Comment:
I don't believe so, an `SQLException` here will mean that the engine couldn't have started correctly in any case. Off the top of my head, the only new possibility would be a connection pool exhaustion, but if it happens at this stage, we'd be having other big issues. I'll do what you mentioned, getting a `null` will skip the check.

Reply:
If we are sure there won't be any side effects, I'm also happy to stick with the initial solution. 👍  I just thought when the data source is already set and not initialized by bootstrapping the engine, this SQL exception would only be thrown on startup. Maybe there are some scenarios where getting the transaction isolation level throws an exception, but interacting with the data source still works fine. This would then be a bummer for existing users, but it is also not all too bad because we don't backport this.

Reply:
Thinking about it, there could eventually be a corner scenario where this particular method could throw an exception where other database operations would not. If we want to stay on the safe side, and given how rare such a scenario would be, I think ignoring it as you suggested could be a good solution instead of stopping the engine because of this.